presto-iceberg/src/main/java/com/facebook/presto/iceberg/PartitionTransforms.java

```diff
@@ -172,11 +247,21 @@ private static Block bucketInteger(Block block, int count)
         return bucketBlock(block, count, position -> bucketHash(INTEGER.getLong(block, position)));
     }
 
+    private static int bucketValueInteger(Block block, int position, int count)
+    {
+        return bucketValue(block, count, position, pos -> bucketHash(INTEGER.getLong(block, pos)));
```

Comment:
Here and for some of the other newly added methods: Is it intended that this calls `bucketValue(..., count, position, ...)` even though the signature of the method is `bucketValue(..., int position, int count, ...)` (that is, the arguments are switched)?

Sorry for this drive-by and late review comment, hopefully it is useful though. I wasn't sure whether a separate issue or discussion was justified.

Reply:
Thanks for pointing this out. I think in some cases the code will still work but bucketing will technically be incorrect. We should switch the arguments. I'll open a PR later today to fix this

Reply:
Nice catch @Marcono1234, thanks for pointing out this, seems we didn't have sufficient test cases to cover these codes. So perhaps we need to supplement some test cases that can cover these code logics in the future.